Xamarin.Forms.Core/Shell/Shell.cs

```diff
@@ -130,6 +130,40 @@ static void OnFlyoutBehaviorChanged(BindableObject bindable, object oldValue, ob
 			BindableProperty.CreateAttached("UnselectedColor", typeof(Color), typeof(Shell), Color.Default,
 				propertyChanged: OnColorValueChanged);
 
+		public static readonly BindableProperty BadgeTextColorProperty =
+			BindableProperty.CreateAttached("BadgeTextColor", typeof(Color), typeof(Shell), Color.Default,
+				propertyChanged: PropagateBadgePropertyChanged(nameof(BaseShellItem.EffectiveBadgeTextColor)));
+
+		public static readonly BindableProperty BadgeUnselectedTextColorProperty =
+			BindableProperty.CreateAttached("BadgeUnselectedTextColor", typeof(Color), typeof(Shell), Color.Default,
+				propertyChanged: PropagateBadgePropertyChanged(nameof(BaseShellItem.EffectiveBadgeTextColor)));
+
+		public static readonly BindableProperty BadgeColorProperty =
+			BindableProperty.CreateAttached("BadgeColor", typeof(Color), typeof(Shell), Color.Default,
+				propertyChanged: PropagateBadgePropertyChanged(nameof(BaseShellItem.EffectiveBadgeColor)));
+
+		public static readonly BindableProperty BadgeUnselectedColorProperty =
+			BindableProperty.CreateAttached("BadgeUnselectedColor", typeof(Color), typeof(Shell), Color.Default,
+				propertyChanged: PropagateBadgePropertyChanged(nameof(BaseShellItem.EffectiveBadgeColor)));
+
+		public static readonly BindableProperty BadgeMoreTextProperty =
+			BindableProperty.CreateAttached("BadgeMoreText", typeof(string), typeof(Shell), null,
+				propertyChanged: PropagateBadgePropertyChanged(nameof(BaseShellItem.EffectiveBadgeMoreText)));
+
+		public static BindableProperty.BindingPropertyChangedDelegate PropagateBadgePropertyChanged(string propertyName) =>
```

Comment:
not required once the properties are moved, and we agree to drop the inheritance part.

Reply:
Dropped.